service/controller/azuremachinepool/handler/nodepool/create_deployment_uninitialized.go

```diff
@@ -124,6 +124,17 @@ func (r *Resource) deploymentUninitializedTransition(ctx context.Context, obj in
 			return currentState, microerror.Mask(err)
 		}
 
+		if !contains(changes, "scaling") {
+			desiredParameters, err := template.NewFromDeployment(desiredDeployment)
+			if err != nil {
+				return currentState, microerror.Mask(err)
+			}
+
+			if desiredParameters.Scaling.MinReplicas == desiredParameters.Scaling.MaxReplicas && desiredParameters.Scaling.CurrentReplicas != int32(*vmss.Sku.Capacity) {
+				changes = append(changes, "scaling")
+			}
```

Comment:
This is actually very clever way to workaround the problem. Did you test that this works in practice?

Reply:
Yes I tested as many use cases I could think of